src/frontend/handlers.go

```diff
@@ -584,19 +539,34 @@ func renderHTTPError(log logrus.FieldLogger, r *http.Request, w http.ResponseWri
 
 	w.WriteHeader(code)
 
-	if templateErr := templates.ExecuteTemplate(w, "error", map[string]interface{}{
+	if templateErr := templates.ExecuteTemplate(w, "error", injectDefaultData(r, map[string]interface{}{
+		"error":       errMsg,
+		"status_code": code,
+		"status":      http.StatusText(code),
+	})); templateErr != nil {
+		log.Println(templateErr)
+	}
+}
+
+func injectDefaultData(r *http.Request, payload map[string]interface{}) map[string]interface{} {
```

Comment:
Love this! 
**Suggestion:** But let's rename this function so that it's more clear.
Some options:
- `injectDefaultTemplateData`
- `injectCommonTemplateData`

(`data` is definitely the correct word here — based on [the documentation](https://pkg.go.dev/html/template#Template.ExecuteTemplate).)

Reply:
Oh yeah that makes more sense. Something about "naming things" being a hard problem :)